James 2843 Restore compatibility with Eclipse

rouazana opened the pull request.

mbaechler commented on json/src/test/java/org/apache/JsonGenericSerializerTest.java:
```diff
@@ -78,6 +80,7 @@ void serializeShouldHandleAllKnownTypes(BaseType domainObject, String serialized
     @ParameterizedTest
     @MethodSource
     void deserializeShouldHandleAllKnownTypes(BaseType domainObject, String serializedJson) throws Exception {
+        @SuppressWarnings({ "rawtypes", "unchecked" })
```
why this one is not treated like the others?

rouazana replied:
fixed, sorry

mbaechler commented on mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java:
```diff
@@ -356,7 +356,7 @@ public ComposedMessageId appendMessage(InputStream msgIn, Date internalDate, fin
                 // https://issues.apache.org/jira/browse/IMAP-122
                 final MimeTokenStream parser = getParser(bIn);
-                final HeaderImpl header = readHeader(parser);
+                readHeader(parser);
```
isn't it a bug?

chibenwa replied:
Probably a side effect...

rouazana replied:
I did not take much time to check it, but it seems the stream is just consumed. But I'm happy it raises concern, because I was a little worried about it.

mbaechler replied:
do we open a ticket for that?

Arsnael commented:
Conflicts with master !

rouazana commented:
rebased and fixed new warnings
I explicetely put the type for the serialization modules, it's a little bloated, so tell me if you prefer that I ignore them

mbaechler commented:
That's ok for me about the explicit type

rouazana commented:
```
[1e44107493cb72a426ed6345a1399a18e36dd5a8] [ERROR] Errors: 
[1e44107493cb72a426ed6345a1399a18e36dd5a8] [ERROR]   JamesMailSpoolerTest.threadSuicideShouldAckTheItem:126 » ConditionTimeout Cond...
```
test this please

rouazana commented:
test this please (temporary broken master)

rouazana commented:
:green_apple: before restarting CI

rouazana commented:
merged
